src/portgraph.rs

```diff
@@ -1284,4 +1523,125 @@ mod test {
         g.remove_node(node1);
         assert!(g.link_ports(node1_output, node0_input).is_err());
     }
+
+    #[test]
+    fn compact_operations() {
+        let mut g = PortGraph::new();
+        let x = g.add_node(3, 2);
+        let a = g.add_node(1, 1);
+        let b = g.add_node(2, 2);
+        let c = g.add_node(1, 1);
+        g.link_nodes(a, 0, b, 0).unwrap();
+        g.link_nodes(b, 0, b, 1).unwrap();
+        g.link_nodes(b, 1, c, 0).unwrap();
+        g.link_nodes(c, 0, a, 0).unwrap();
+
+        assert_eq!(g.link_count(), 4);
+        assert_eq!(g.node_count(), 4);
+        assert_eq!(g.port_count(), 13);
+
+        g.remove_node(x);
+
+        assert_eq!(g.link_count(), 4);
+        assert_eq!(g.node_count(), 3);
+        assert_eq!(g.port_count(), 8);
+        assert!(g.connected(a, b));
+        assert!(g.connected(b, b));
+        assert!(g.connected(b, c));
+        assert!(g.connected(c, a));
+
+        let mut new_nodes = HashMap::new();
+        g.compact_nodes(|old, new| {
+            new_nodes.insert(old, new);
+        });
+
```

Comment:
```suggestion
assert_neq!(a, new_nodes[&a])
```
and so on? (for whichever ones we actually expect to change)

Reply:
I'm not sure we want to check strict non-equality, I added more global tests instead (collecting into vectors and checking equality there).